### PR TITLE
Verdi shell migration to click

### PR DESCRIPTION
**Pull request by szoupanos**
Solves issue #1594 
Please, let me know if something more is needed.

Thanks,
Spyros

### CODE REVIEW

File: setup_requirements.py [hunk 1]
```diff
@@ -46,6 +46,7 @@
     'plumpy==0.10.2',
     'circus==0.14.0',
     'tornado==4.5.3',  # As of 2018/03/06 Tornado released v5.0 which breaks circus 0.14.0
+    'bpython==0.17.1',
```
Comment (sphuber) [inline setup_requirements.py]: Should this be a new standard requirement? And if so, remember to add it to rtd requirements

Comment (giovannipizzi) [reply setup_requirements.py]: Indeed, I wouldn't add bpython (we just reused the logic of Django, at least in some old version, that was allowing also for bpython if ipython was not there, but I'm not even sure of if bpython still exists...)

Comment (szoupanos) [reply setup_requirements.py]: It exists. I installed it to test this. I can remove the dependency but for me it doesn't make sense to try to load a shell in AiiDA (and also to explicitly mention it at the AiiDA cli arguments) when it is not part of the AiiDA dependencies.

**Comment by szoupanos** [comment]
And regarding the tests, I have to think. I am not sure how this can be tested either.
(I am open to suggestions)

File: setup_requirements.py (hunk 1 above)
Comment (DropD) [reply setup_requirements.py]: I would say it should be an extra requirement.

File: aiida/cmdline/commands/shell.py [hunk 2]
```diff
@@ -35,183 +37,177 @@
     # ("aiida.backends.sqlalchemy", "models", "models"),
 ]
 
+
 class Shell(VerdiCommand):
     """
-    Run the interactive shell with the AiiDA environment loaded.
-
-    This command opens an ipython shell with the AiiDA environment loaded.
+    Re-hash nodes filtered by identifier and or node class
```
[the remark this answers is not in the document]
Comment (DropD) [reply aiida/cmdline/commands/shell.py]: fixed in my upcoming PR

File: aiida/cmdline/commands/shell.py [hunk 3]
```diff
@@ -35,183 +37,177 @@
...
+    import bpython
+
+    user_ns = get_start_namespace()
+    if user_ns:
+        bpython.embed(user_ns)
+    else:
+        bpython.embed()
+
+shells = {'ipython': ipython , 'bpython': bpython}
+@verdi_shell.   command('shell')
```
[the remark this answers is not in the document]
Comment (DropD) [reply aiida/cmdline/commands/shell.py]: fixed in my upcoming PR

File: aiida/cmdline/commands/shell.py [hunk 4]
```diff
@@ -35,183 +37,177 @@
...
+@click.option(
+    '--plain', is_flag=True,
+    help='Tells Django to use plain Python, not IPython or bpython.)'
+)
+@click.option(
+    '--no-startup', is_flag=True,
+    help='When using plain Python, ignore the PYTHONSTARTUP environment '
+         'variable and ~/.pythonrc.py script.'
+)
+@click.option('-i', '--interface', type=click.Choice(shells.keys()),
```
[the remark this answers is not in the document]
Comment (DropD) [reply aiida/cmdline/commands/shell.py]: fixed in my upcoming PR

**Comment by giovannipizzi** [comment]
Superseded by #1732

**Pull request closed without merging**